### PR TITLE
[CDF-987] XSS vulnerability in Date Range Input Component

### PR DESCRIPTION
**Pull request by Advent51**
- input value sets through jquery

### CODE REVIEW

**Comment by Advent51** [comment]
@pamval @mbatchelor @mchen-len-son please review

**Comment by afrjorge** [comment]
Maybe a test should be added for this. [CDF-985](http://jira.pentaho.com/browse/CDF-985) is similar, see what was done to the test file [TextInputComponent-spec.js](https://github.com/webdetails/cdf/blob/master/core-js/src/test/javascript/cdf/components/TextInputComponent-spec.js#L68)

**Comment by Advent51** [comment]
@afrjorge @annapopova1 @mchen-len-son @pamval please review

File: core-js/src/main/javascript/cdf-legacy/components/input.js [hunk 1]
```diff
@@ -699,16 +699,23 @@ var DateRangeInputComponent = BaseComponent.extend({
...
+      dr = $('<input class="date-range-multiple-input" id="' + inputId + '"/>' + inputSeparator +
+      '<input class="date-range-multiple-input-2" id="' + inputId + '2' + '"/>');
     }
 
     this.placeholder()
         .addClass('date-range-input-container')
         .html(dr);
 
+    if (this.singleInput == undefined || this.singleInput == true) {
+      $('#' + inputId).val(startValue + ' ' + inputSeparator + ' ' + endValue);
```
Comment (afrjorge) [inline core-js/src/main/javascript/cdf-legacy/components/input.js]: Using selectors to search for elements that are being created here seems to be excessive and might affect performance.

**Comment by Advent51** [comment]
@mbatchelor @pamval please review

**Pull request merged**